src/eval.ml

```diff
@@ -187,34 +193,44 @@ and tree_walk : dtree -> stack -> (term * stack) option = fun tree stk ->
                 env.(slot) <- TE_Some(b)
               else
                 let b = lift vars.(pos) in
-                let xs = Array.map (fun e -> VarMap.find e fresh_vars) xs in
+                let xs = Array.map (fun e -> IntMap.find e id_vars) xs in
                 env.(slot) <- TE_Some(Bindlib.unbox (Bindlib.bind_mvar xs b))
         in
         List.iter fn env_builder;
         (* Actually perform the action. *)
         Some(Bindlib.msubst act env, stk)
-    | Cond({ ok ; cond ; fail })                          ->
+    | Cond({ok; cond; fail})                              ->
         let next =
           match cond with
           | CondNL(i, j) ->
               if eq_modulo vars.(i) vars.(j) then ok else fail
           | CondFV(xs,i) ->
-              let xs = Array.map (fun e -> VarMap.find e fresh_vars) xs in
+              let allowed = Array.map (fun e -> IntMap.find e id_vars) xs in
+              (* FIXME some comparisons can be avoided. *)
```

Comment:
I remember we discussed this together but, for the record, it would be nice to add some example.

Reply:
The question is: is an `Array.mem` faster than a `Bindlib.occur`?

Reply:
Could you please remind me the example we discussed together? I cannot remember it right now.

Reply:
> The question is: is an Array.mem faster than a Bindlib.occur?

Should be roughly the same.

Reply:
@fblanqui, the current code evinces the redundancy, it concerns the variables held by the condition node

Reply:
As of 7ad8397826e3ebc80ad17acb2175912628ad6ca9, the map on which we iterate to check for variable occurrence is freed from the variables held by the constraint